fix issue#130 cl ls, cl wls, cl search return structured_info

pujun-ai opened the pull request:
### Change

Fixes #130 Now cl search, cl wls and cl ls return structured information
### Testing
#### Unit tests

```
$ nosetests
 BundleStore.upload: moving test_root/temp/abloogywoogywu to test_root/data/0xdirectory-hash
.....................
----------------------------------------------------------------------
Ran 28 tests in 5.834s

OK

```
#### Manual tests

![something](https://cloud.githubusercontent.com/assets/5567951/10125548/97153f54-652d-11e5-912e-80b734bc6812.png)

Commit:
fix#130: cl ls, cl wls, cl search return structured_info

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -812,6 +812,24 @@ def do_search_command(self, argv, parser):
             print 'Added %d bundles to %s' % (len(bundle_uuids), self.worksheet_str(worksheet_info))
+        return self.create_structured_info_map([('refs', reference_map),])
+
+    def create_structured_info_map(self, structured_info_list):
```
Add a comment on what the purpose of this structured information is...

percyliang commented on codalab/lib/bundle_cli.py:
```diff
@@ -812,6 +812,24 @@ def do_search_command(self, argv, parser):
+        return { k: v for k, v in structured_info_list}
+
+    def shorten_uuid(self, uuid):
+        return uuid[0:8]
```
Coordinate this with the place that actually shortens to 8 characters

Commit:
fix#130: additional information returned from methods do_wls, do_ls, do_search

pujun-ai commented:
Changes modifications:
    added comments to explain the use of  additional_information in reconstructing URLs. 
    Removed shorten_url logic and replaced it with worksheet_util.apply_func.

percyliang commented:
Looks good to me.

On Sun, Sep 27, 2015 at 6:51 PM, Pujun Bhatnagar notifications@github.com
wrote:

> Changes modifications:
> added comments to explain the use of additional_information in
> reconstructing URLs.
> Removed shorten_url logic and replaced it with worksheet_util.apply_func.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codalab/codalab-cli/pull/147#issuecomment-143620239.

kashizui commented:
Will review this today!

kashizui commented on codalab/lib/bundle_cli.py:
```diff
@@ -812,6 +812,31 @@ def do_search_command(self, argv, parser):
+        which are too short. This dict contains additional information that is
+        needed to recover URL on the client side.
+        '''
+        return { k: v for k, v in structured_info_list}
```
nit: use even spacing on either size of comprehension (I would use no space at all).

kashizui replied:
(I'll use "nit:" from now on to indicate that I'm just being nitpicky and you're ok to merge without making the change.)

kashizui replied:
You can also do the same with the dict constructor without the comprehension: `dict(structured_info_list)`

kashizui commented on codalab/lib/bundle_cli.py:
```diff
@@ -825,6 +850,8 @@ def do_ls_command(self, argv, parser):
         if len(bundle_info_list) > 0:
             self.print_bundle_info_list(bundle_info_list, args.uuid_only, print_ref=True)
+        reference_map = self.create_reference_map('bundle', bundle_info_list)
+        return self.create_structured_info_map([('refs', reference_map),])
```
nit: extra comma not necessary (to match L1418)

kashizui commented on codalab/lib/bundle_cli.py:
```diff
@@ -812,6 +812,31 @@ def do_search_command(self, argv, parser):
                 client.add_worksheet_item(worksheet_uuid, worksheet_util.bundle_item(bundle_uuid))
             worksheet_info = client.get_worksheet_info(worksheet_uuid, False)
             print 'Added %d bundles to %s' % (len(bundle_uuids), self.worksheet_str(worksheet_info))
+        return self.create_structured_info_map([('refs', reference_map),])
```
nit: extra comma not necessary (to match L1418)

kashizui commented on codalab/lib/bundle_cli.py:
```diff
@@ -812,6 +812,31 @@ def do_search_command(self, argv, parser):
+        in the info_list. This information is needed to recover URL on the cient side.
+        '''
+        if len(info_list) == 0:
+            return None
```
Is there any reason to return `None` rather than an empty dict?

pujun-ai replied:
This was mostly a design decision. An empty dictionary could be returned but I decided to go with a None since that can be easily be checked wherever this method is called and one does not have find the length of dictionary to know that there is no information present.

percyliang replied:
I'd use {} for consistency to avoid extra checks (len is basically free).

On Fri, Oct 2, 2015 at 10:45 PM, Pujun Bhatnagar notifications@github.com
wrote:

> In codalab/lib/bundle_cli.py
> https://github.com/codalab/codalab-cli/pull/147#discussion_r41082934:
> 
> > -    def create_structured_info_map(self, structured_info_list):
> > -        '''
> > -        Return dict of info dicts (eg. bundle/worksheet reference_map) containing
> > -        information associated to bundles/worksheets. cl wls, ls, etc. show uuids
> > -        which are too short. This dict contains additional information that is
> > -        needed to recover URL on the client side.
> > -        '''
> > -        return { k: v for k, v in structured_info_list}
> >   +
> > -    def create_reference_map(self, info_type, info_list):
> > -        '''
> > -        Return dict of dicts containing name, uuid and type for each bundle/worksheet
> > -        in the info_list. This information is needed to recover URL on the cient side.
> > -        '''
> > -        if len(info_list) == 0:
> > -            return None
> 
> This was mostly a design decision. An empty dictionary could be returned
> but I decided to go with a None since that can be easily be checked
> wherever this method is called and one does not have find the length of
> dictionary to know that there is no information present.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codalab/codalab-cli/pull/147/files#r41082934.

kashizui commented:
Logic looks good to me!

Commit:
fix #130 cl wls, ls, search returns more info
